Week2/TriplateSum.java

```diff
@@ -0,0 +1,81 @@
+package sum3;
+
+import java.util.*;
+
+public class TriplateSum 
+{
+	public static Set<Set<Integer>> Sum3(int ar[])
+	{
+		 Set<Set<Integer>> bags = new HashSet<>();
+		 int ans[]= {0},n=ar.length,left,right;
+		 if(ar == null || ar.length<3) //array is NULL or less than two elements;
+			 return null;
+		 Arrays.sort(ar); //sorting array
+		 for(int i=0;i<n-2;i++) //traversing array for top n-2 elements 
+		 {
+			 left=i+1; //next bigger element from left
+			 right=n-1;// most largest element
+		 
+		  while(left<right) //till elements indices don't coincide
+		  {
+			  int sum=ar[i]+ar[left]+ar[right];
+			  if(sum==0) // check wheather they are eligible to make ZERO or not ;
+			  {
+				 // System.out.println("<"+ar[i]+","+ar[left]+","+ar[right]+">");
+				  
+				  SortedSet<Integer> temp=new TreeSet<>(); //new temp list

```

Comment:
Why do we need a SortedSet here? Wouldn't that prevent a solution such as [-1, -1, 2] from being valid because a Set would not let you add duplicate values

Reply:
i used sortedset for sorting elements in each triplet... hmm it will prevent. I will alter Sortedset to normal list with sorting order.